pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/channel/ServiceUnitState.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.broker.loadbalance.extensions.channel;
+
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Defines the possible states for service units.
+ *
+ * The following diagram defines the valid state changes
+ *
+ *                  ┌───────────┐
+ *       ┌──────────┤ released  │◄────────┐
+ *       │own       └───────────┘         │release
+ *       │                                │
+ *       │                                │
+ *       ▼                                │
+ *    ┌────────┐  assign(transfer)  ┌─────┴────┐
+ *    │        ├───────────────────►│          │
+ *    │ owned  │                    │ assigned │
+ *    │        │◄───────────────────┤          │
+ *    └──┬─────┤      own           └──────────┘
+ *       │  ▲  │                         ▲
+ *       │  │  │                         │
+ *       │  │  └──────────────┐          │
+ *       │  │                 │          │
+ *       │  │        unload   │          │ assign(assignment)
+ * split │  │                 │          │
+ *       │  │                 │          │
+ *       │  │ create(child)   │          │
+ *       │  │                 │          │
+ *       ▼  │                 │          │
+ *    ┌─────┴─────┐           └─────►┌───┴──────┐
+ *    │           │                  │          │
+ *    │ splitting ├────────────────► │   free   │
+ *    │           │   discard(parent)│          │
+ *    └───────────┘                  └──────────┘
+ */
+public enum ServiceUnitState {
```

Comment:
Is this `ServiceUnit` the same as `bundle` for now and in the foreseeable future?

Reply:
I would like to keep open the door to having also a single topic (partition) and not only a "bundle"

Reply:
I see two conflicting comments here.

The intention is to keep this interface open for other service units(topic, etc). But the initial target service units will be bundles only.

Reply:
> The intention is to keep this interface open for other service units(topic, etc). But the initial target service units will be bundles only.

Does support for other "service units" included in PIP-192?

Both is fine for me, as long as we don't mixing them up. If we stick to service unit, so there won't be any 'bundle' appear in load balance module unless it's some specific implementation, right?

Reply:
For pip-192, no, it will be bundle only. 

Sure, we can stick to "serviceUnit" for var and func names.